Engine/GameObject.cpp

```diff
@@ -37,13 +37,13 @@ GameObject::~GameObject()
 		parent->RemoveChild(this);
 	}
 
-	for (unsigned int i = 0; i < components.size(); ++i)
+	for (int i = (components.size() - 1); i >= 0; --i)
```

Comment:
Why in inverse order?

Reply:
When deleting stuff from a list/vector I prefer to remove it in inverse order to avoid problems. This way you make sure you are going to iterate trought all the elements even if the size of the list/vector change